samtranslator/model/api/api_generator.py

```diff
@@ -667,6 +805,17 @@ def _construct_alias_target(self, domain: Dict[str, Any], api_domain_name: str,
             alias_target["DNSName"] = route53.get("DistributionDomainName")
         return alias_target
 
+    def _create_basepath_mapping(
```

Comment:
call it `_v2`, because in theory you could be doing the same for v1, but this one explicitly uses v2.

(It would be good if you actually create another method that does this for v1 and use it in the other version)